src/Approximators/RangeApproximators/rangefinder.jl

```diff
@@ -0,0 +1,150 @@
+    ``(AA^\\top)^q AS``, this is known as taking ``q`` power iterations. Power iterations 
+    drive the constant on ``\\sigma_{k+1}`` in the bound closer to 1, leading to more 
+    accurate approximations. One can also improve the stability of these power iterations
+    be orthogonalizing each matrix in what is known as the random subspace iteration.
+
+# Fields
+- `compressor::Compressor`, the technique that will compress the matrix from the right.
+- `power_its::Int64`, the number of power iterations that should be performed.
+- `rand_subspace::Bool`, a boolean indicating whether the power its should be performed 
+    with orthogonalization.
+"""
+mutable struct RangeFinder <: RangeApproximator
+    compressor::Compressor
+    power_its::Int64
+    rand_subspace::Bool
+    function RangeFinder(compressor, power_its, rand_subspace)
+        if power_its < 0
+            throw(ArgumentError("Field `power_its` must be non-negative."))
+        end
+        
+        return new(compressor, power_its, rand_subspace)
+    end
+
+end
+
+RangeFinder(;
+    compressor = SparseSign(), 
+    rand_subspace = false, 
+    power_its = 1
+   ) = RangeFinder(compressor, rand_subspace, power_its)
+
+"""
+    RangeFinderRecipe
+
+A struct that contains the preallocated memory and completed compressor to form a
+    RangeFinder approximation to the matrix ``A``.
+
+# Fields
+- `compressor::CompressorRecipe`, the compressor to be applied from the right to ``A``.
+- `power_its::Int64`, the number of power iterations that should be performed.
+- `rand_subspace::Bool`, a boolean indicating whether the power its should be performed 
+    with orthogonalization.
+- `range::AbstractMatrix`, the orthogonal matrix that approximates the range of ``A``.
+"""
+mutable struct RangeFinderRecipe <: RangeApproximatorRecipe
+    n_rows::Int64
+    n_cols::Int64
+    compressor::CompressorRecipe
+    power_its::Int64
+    rand_subspace::Bool
+    range::AbstractMatrix
+end
+
+function complete_approximator(approx::RangeFinder, A::AbstractMatrix)
+    type = eltype(A)
+    # You need to make sure you orient the compressor in the correct direction
+    if typeof(approx.compressor.cardinality) <: Left
+        approx.compressor.cardinality = Right()
+    end
+
+    compress = complete_compressor(approx.compressor, A)
+    # Determine the dimensions of the range approximator
+    a_rows = size(A, 1)
+    c_cols = size(compress, 2)
+    approx_recipe = RangeFinderRecipe(
+        a_rows,
+        c_cols,
+        compress, 
+        approx.power_its,
+        approx.rand_subspace, 
+        Matrix{type}(undef, 2, 2)
+    )
+end
+
+function rapproximate!(approx::RangeFinderRecipe, A::AbstractMatrix)
+    # we don't dispatch on this incase someone wishes to make multiple runs with the 
+    # same recipe
+    if approx.rand_subspace 
+        approx.range = rand_power_it(A, approx)
+    else
+        approx.range = rand_subspace_it(A, approx)
+    end
```

Comment:
Wouldn't this be two distinct recipes? We likely want to keep recipes immutable (note, pointers to arrays that are modified in place should not be a problem here.)